R/plot.R

```diff
@@ -179,14 +179,15 @@ embeddingPlot <- function(embedding, groups=NULL, colors=NULL, plot.na=TRUE, min
     geom_point_w <- ggplot2::geom_point
   }
 
-  if (!is.null(groups)) {
+  if (!is.null(groups) && is.null(colors)) {
     groups <- as.factor(groups)
 
     plot.df$Group <- factor(NA, levels=levels(groups))
     arr.ids <- match(names(groups), plot.df$CellName)
     plot.df$Group[arr.ids[!is.na(arr.ids)]] <- groups[!is.na(arr.ids)]
 
-    big.clusts <- (plot.df %>% dplyr::group_by(Group) %>% dplyr::summarise(Size=n()) %>%
+
+    big.clusts <- (plot.df %>% subset(!is.na(Group)) %>% dplyr::group_by(Group) %>% dplyr::summarise(Size=n()) %>%
```

Comment:
Hm, did you test this? I'm surprised that it works, as `subset` is not dplyr function, and it shouldn't process raw column names, i.e. I'd expect it should be `plot.df %>% subset(!is.na(.$Group))` or `plot.df %>% dplyr::filter(!is.na(Group))`. But if it works, then fine.

Reply:
Just double-checked:

`> head(plot.df)
                         CellName            x         y   Group
1 ctrl_037_one_AGAGCAGCAACCAGAG-1  -3.13941365 -39.59650 Neurons
2 ctrl_039_one_AAACCCAAGATGGCAC-1   5.96256323 -43.23856 Neurons
3 ctrl_037_one_AGATAGAAGACTAAGT-1  -9.79544731 -39.37908 Neurons
4 ctrl_037_one_AGCCACGGTCAAGGCA-1 -33.28515343 -32.66608     OPC
5 ctrl_037_one_AGTTAGCAGTACGAGC-1 -34.02428791 -32.12197     OPC
6 ctrl_037_one_ATTTACCTCTCCCATG-1   0.08197635 -39.58852 Neurons
> str(plot.df)
'data.frame':	75915 obs. of  4 variables:
 $ CellName: chr  "ctrl_037_one_AGAGCAGCAACCAGAG-1" "ctrl_039_one_AAACCCAAGATGGCAC-1" "ctrl_037_one_AGATAGAAGACTAAGT-1" "ctrl_037_one_AGCCACGGTCAAGGCA-1" ...
 $ x       : num  -3.14 5.96 -9.8 -33.29 -34.02 ...
 $ y       : num  -39.6 -43.2 -39.4 -32.7 -32.1 ...
 $ Group   : Factor w/ 6 levels "Glia","Astrocytes",..: 3 3 3 4 4 3 4 3 4 4 ...
> plot.df %<>% subset(!is.na(Group))
> str(plot.df)
'data.frame':	75877 obs. of  4 variables:
 $ CellName: chr  "ctrl_037_one_AGAGCAGCAACCAGAG-1" "ctrl_039_one_AAACCCAAGATGGCAC-1" "ctrl_037_one_AGATAGAAGACTAAGT-1" "ctrl_037_one_AGCCACGGTCAAGGCA-1" ...
 $ x       : num  -3.14 5.96 -9.8 -33.29 -34.02 ...
 $ y       : num  -39.6 -43.2 -39.4 -32.7 -32.1 ...
 $ Group   : Factor w/ 6 levels "Glia","Astrocytes",..: 3 3 3 4 4 3 4 3 4 4 ...
> head(plot.df)
                         CellName            x         y   Group
1 ctrl_037_one_AGAGCAGCAACCAGAG-1  -3.13941365 -39.59650 Neurons
2 ctrl_039_one_AAACCCAAGATGGCAC-1   5.96256323 -43.23856 Neurons
3 ctrl_037_one_AGATAGAAGACTAAGT-1  -9.79544731 -39.37908 Neurons
4 ctrl_037_one_AGCCACGGTCAAGGCA-1 -33.28515343 -32.66608     OPC
5 ctrl_037_one_AGTTAGCAGTACGAGC-1 -34.02428791 -32.12197     OPC
6 ctrl_037_one_ATTTACCTCTCCCATG-1   0.08197635 -39.58852 Neurons`